x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypes.java

```diff
@@ -151,6 +152,10 @@ public static boolean isSigned(DataType t) {
         return t.isNumeric();
     }
 
+    public static boolean isArray(DataType t) {
+        return t instanceof ArrayDataType;
```

Comment:
I'd rather have the isArray() as a class mathod (not static) where the basic data type returns false and array true, so we avoid `nstanceof`

Reply:
That's also a possibility, indeed.
During an earlier analysis with Costin we discussed if non-array types should be aware of the array properties (my initial proposal was for DateType to have the `baseType` field and `isArray` method). The outcome however was to subclass (and thus `ArrayDataType` appeared).
A mixed approach would be to add the `isArray()` method in `DataType` always return `false` and in `ArrayDataType` always return `true`, but that wouldn't be a strict separation between the types anymore. But doable, still.

Reply:
`instanceof` has some performance impact, at least in the past jvms :), but it's also a personal preference to be honest, I prefer methods like the `isArray()`, they seem more elegant to me.